### PR TITLE
feat: add feature toggles to waiting lists

### PR DESCRIPTION
**Pull request by AxonC**

### CODE REVIEW

File: app/Models/Training/WaitingList.php [hunk 1]
```diff
@@ -175,6 +179,28 @@ public function getFormattedDepartmentAttribute()
         };
     }
 
+    public function getShouldCheckAtcHoursAttribute(): bool
+    {
+        // default to true
+        return $this->isAtcList() && ($this->feature_toggles['check_atc_hours'] ?? true);
```
Comment (CalumTowers) [inline app/Models/Training/WaitingList.php:185]: Why is this not just looking at the toggle? Would be much cleaner and obvious to the user as to what is going on. Also would remove the need for the comment.

File: app/Models/Training/WaitingList.php [hunk 2]
```diff
@@ -175,6 +179,28 @@ public function getFormattedDepartmentAttribute()
...
+    }
+
+    public function getShouldCheckCtsTheoryExamAttribute(): bool
+    {
+        // default to true for atc lists.
+        return $this->isAtcList() && ($this->feature_toggles['check_cts_theory_exam'] ?? true);
```
Comment (CalumTowers) [inline app/Models/Training/WaitingList.php:191]: Why is this not just looking at the toggle? Would be much cleaner and obvious to the user as to what is going on. Also would remove the need for the comment.

File: app/Models/Training/WaitingList.php [hunk 3]
```diff
@@ -175,6 +179,28 @@ public function getFormattedDepartmentAttribute()
...
+    }
+
+    public function getFeatureTogglesFormattedAttribute(): object
+    {
+        // convert feature toggles to an object with defaults included to
+        // ensure keys are always present.
```
Comment (CalumTowers) [inline app/Models/Training/WaitingList.php:197]: I'm not sure this comment is helpful, the code is fairly verbose

File: app/Models/Training/WaitingList/WaitingListAccount.php [hunk 4]
```diff
@@ -140,6 +140,10 @@ public function atcHourCheck()
             return true;
         }
 
+        if ($this->waitingList->should_check_atc_hours == false) {
```
Comment (CalumTowers) [inline app/Models/Training/WaitingList/WaitingListAccount.php:143]:
```suggestion
        if (! $this->waitingList->should_check_atc_hours) {
```

**Review by CalumTowers — CHANGES_REQUESTED** [review]

File: app/Models/Training/WaitingList.php (hunk 1 above)
Comment (AxonC) [reply app/Models/Training/WaitingList.php:185]: Historically we have only shown cts theory exams & atc hour checks on ATC lists so this is just maintaining backwards compatibility.

File: app/Models/Training/WaitingList.php (hunk 2 above)
Comment (AxonC) [reply app/Models/Training/WaitingList.php:189]: changed

File: app/Models/Training/WaitingList.php (hunk 1 above)
Comment (AxonC) [reply app/Models/Training/WaitingList.php:184]: changed

**Pull request merged**